src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol_ImplementationChecks.cs

```diff
@@ -1058,6 +1072,29 @@ static bool isOrContainsErrorType(TypeSymbol typeSymbol)
             }
         }
 
+        /// <summary>
+        /// Return true if <paramref name="overridingReturnType"/> is valid for the return type of an override method when the overridden method's return type is <paramref name="overriddenReturnType"/>.
+        /// </summary>
+        private bool IsValidOverrideReturnType(Symbol overridingSymbol, TypeWithAnnotations overridingReturnType, TypeWithAnnotations overriddenReturnType, DiagnosticBag diagnostics)
+        {
+            if (DeclaringCompilation.LanguageVersion >= MessageID.IDS_FeatureCovariantReturnsForOverrides.RequiredVersion())
+            {
+                HashSet<DiagnosticInfo> useSiteDiagnostics = null;
+                var result = DeclaringCompilation.Conversions.HasIdentityOrImplicitReferenceConversion(overridingReturnType.Type, overriddenReturnType.Type, ref useSiteDiagnostics);
```

Comment:
>HasIdentityOrImplicitReferenceConversion [](start = 62, length = 40)

This would allow overriding with a derived interface, no?
From the runtime PR, I see that's not going to be allowed: "Return types in covariant return methods can only be reference types: covariant interface return types are not supported."

Reply:
It can be any implicit reference conversion. What is not supported is covariant overrides of methods declared in interfaces.

---
In reply to: [414047290](https://github.com/dotnet/roslyn/pull/43576#discussion_r414047290) [](ancestors = 414047290)

Reply:
Should the description in runtime PR be corrected then?

---
In reply to: [414049729](https://github.com/dotnet/roslyn/pull/43576#discussion_r414049729) [](ancestors = 414049729,414047290)

Reply:
I've started an email thread about that question.

---
In reply to: [414050605](https://github.com/dotnet/roslyn/pull/43576#discussion_r414050605) [](ancestors = 414050605,414049729,414047290)